src/components/split_button/split_button.tsx

```diff
@@ -0,0 +1,225 @@
+     */
+    itemClassName?: string;
+
+    /**
+     * optional index of options item to mark with checkmark
+     */
+    selectedIndex?: number;
+
+    /**
+     * This callback will be executed when an option is clicked, in addition to possible
+     * href/onClick of the option-item itself.
+     */
+    onSelection?: (index: number) => void;
+
+    /**
+     * Change to `true` if you want horizontal lines between options.
+     * This is best used when options are multi-line.
+     */
+    hasDividers?: boolean;
+
+    /**
+     * Applied to the outermost wrapper (popover)
+     */
+    popoverClassName?: string;
+
+    /**
+     * Controls whether the options are shown upon initial render. Default: false
+     */
+    initiallyOpen?: boolean;
+
+    /**
+     * Optional additional props to send to Primary Button
+     */
+    propsForPrimaryButton?: OuiButtonProps;
+
+    /**
+     * Optional additional props to send to Dropdown Button
+     */
+    propsForDropdownButton?: OuiButtonProps;
+
+    /**
+     * Optional additional props to send to each Option Item, when
+     * it is string, rendered with OuiText wrapper
+     */
+    propsForOptionItems?: OuiTextProps;
+  };
+
+export const OuiSplitButton = ({
+  color = 'primary',
+  fullWidth = false,
+  disabled,
+  options = [],
+  selectedIndex,
+  onSelection,
+  initiallyOpen = false,
+  hasDividers,
+  itemClassName,
+  onClick,
+  className,
+  popoverClassName,
+  children,
+  propsForDropdownButton,
+  propsForOptionItems,
+  propsForPrimaryButton,
+  ...rest
+}: OuiSplitButtonProps) => {
+  const [isOpen, setIsOpen] = useState(!!initiallyOpen);
+
+  const onSelectKeyDown = (
+    event: ExclusiveUnion<
+      React.KeyboardEvent<HTMLButtonElement>,
+      React.KeyboardEvent<HTMLAnchorElement>
+    >
+  ) => {
+    if (event.key === keys.ARROW_UP || event.key === keys.ARROW_DOWN) {
+      event.preventDefault();
+      event.stopPropagation();
+      setIsOpen(true);
+    }
+  };
+
+  const popoverClasses = classNames('ouiSplitButton', popoverClassName);
+
+  const buttonClasses = classNames(
+    {
+      'ouiSplitButton--isOpen__button': isOpen,
+    },
+    className
+  );
+
+  const itemClasses = classNames(
+    'ouiSplitButton__item',
+    {
+      'ouiSplitButton__item--hasDividers': hasDividers,
+    },
+    itemClassName
+  );
+
+  const button = (
+    <OuiSplitButtonControl
+      color={color}
+      onDropdownClick={() => setIsOpen(!isOpen)}
+      onClick={onClick}
+      onKeyDown={onSelectKeyDown}
+      className={buttonClasses}
+      fullWidth={fullWidth}
+      disabled={disabled}
+      propsForDropdownButton={propsForDropdownButton}
+      propsForPrimaryButton={propsForPrimaryButton}
+      {...rest}>
+      {children}
+    </OuiSplitButtonControl>
+  );
+
+  const textColor = (color: OuiSplitButtonColor) => {
+    if (color === 'primary') return 'success';
+    if (color === 'text') return 'default';
+
+    return color;
+  };
```

Comment:
I feel like the option text color should be a generic text color, not specific to the color of the button.

cc @kgcreative

Reply:
+1

Reply:
@BSFishy @kgcreative @ashwin-pc
This is another discussion point.   Suggestion has been made on both sides.

What principles do we want to apply here to come to a single choice?

To re-state the question :

Should the default option-item text follow the current button-color, or always be "generic" color?